pkg/query/storeset.go

```diff
@@ -341,16 +344,30 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 	s.storesStatusesMtx.Lock()
 	defer s.storesStatusesMtx.Unlock()
 
+	var status StoreStatus
 	now := time.Now()
-	s.storeStatuses[store.addr] = &StoreStatus{
-		Name:      store.addr,
-		LastError: err,
-		LastCheck: now,
-		Labels:    store.labels,
-		StoreType: store.storeType,
-		MinTime:   store.minTime,
-		MaxTime:   store.maxTime,
+
+	if previousStatus, ok := s.storeStatuses[store.addr]; ok {
```

Comment:
I don't really like the fact that we identify store by address - any rollout can change that easily and for a minute we have confusing view.. But I think for UI it's explicit enough, so let's leave it (:

Reply:
What do you think we could replace this by ? Maybe a random ID generated by the store at start up time and send in gRPC info call ?

Reply:
Perhaps we could generate a ULID at startup, pass it here: https://github.com/improbable-eng/thanos/blob/master/pkg/cluster/cluster.go#L161, and then add it to `InfoResponse`, and then use it here. WDYT?

Reply:
That's what I had in mind. Better to probably leave it for another PR ?

What other uses could it have ?

Reply:
Interesting idea, why not (: